resource-service/src/unit-tests/java/uk/gov/gchq/palisade/service/resource/service/SimpleResourceServiceTest.java

```diff
@@ -20,70 +20,51 @@
 
 import uk.gov.gchq.palisade.resource.LeafResource;
 import uk.gov.gchq.palisade.resource.impl.DirectoryResource;
-import uk.gov.gchq.palisade.resource.impl.FileResource;
 import uk.gov.gchq.palisade.util.ResourceBuilder;
 
 import java.io.File;
 import java.io.IOException;
 import java.net.URI;
-import java.util.Optional;
-import java.util.Set;
-import java.util.function.Function;
-import java.util.stream.Collectors;
-import java.util.stream.Stream;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
-public class SimpleResourceServiceTest {
+class SimpleResourceServiceTest {
     private final SimpleResourceService service = new SimpleResourceService("data-service", "java.lang.String");
 
     @Test
-    public void testJavaFilesInSrc() throws IOException {
+    void testJavaFilesInUnitTest() throws IOException {
         // Given
-        Set<LeafResource> javaFiles = service.getResourcesBySerialisedFormat("java").collect(Collectors.toSet());
-        DirectoryResource srcMainJava = (DirectoryResource) ResourceBuilder.create(new File("./src/main/java").getCanonicalFile().toURI());
         DirectoryResource unitTestJava = (DirectoryResource) ResourceBuilder.create(new File("./src/unit-tests/java").getCanonicalFile().toURI());
-        DirectoryResource compTestJava = (DirectoryResource) ResourceBuilder.create(new File("./src/component-tests/java").getCanonicalFile().toURI());
-        DirectoryResource ctractTestJava = (DirectoryResource) ResourceBuilder.create(new File("./src/contract-tests/java").getCanonicalFile().toURI());
 
         // When
-        Stream<LeafResource> sourceFiles = service.getResourcesById(srcMainJava.getId());
-        Stream<LeafResource> testFiles = Stream.of(
-                service.getResourcesById(unitTestJava.getId()),
-                service.getResourcesById(compTestJava.getId()),
-                service.getResourcesById(ctractTestJava.getId()))
-                .flatMap(Function.identity());
-        Set<LeafResource> srcAndTestJavaFiles = Stream.concat(sourceFiles, testFiles).collect(Collectors.toSet());
+        FunctionalIterator<LeafResource> testFiles = FunctionalIterator.fromIterator(service.getResourcesById(unitTestJava.getId()));
 
         // Then
-        assertThat(javaFiles).isEqualTo(srcAndTestJavaFiles);
+        assertThat(testFiles.hasNext()).isTrue();
```

Comment:
Prefer to test equality rather than just "there's stuff"

Reply:
Added a test file to perform a check that the returned entity contains the resource.